source/solvers/heat_transfer.cc

```diff
@@ -754,6 +760,26 @@ HeatTransfer<dim>::attach_solution_to_output(DataOut<dim> &data_out)
     }
 }
 
+template <int dim>
+double
+HeatTransfer<dim>::calculate_delta_T_ref()
+{
+  double previous_solution_maximum, previous_solution_minimum;
+  if (is_steady(simulation_parameters.simulation_control.method))
+    {
+      previous_solution_maximum = this->present_solution.max();
+      previous_solution_minimum = this->present_solution.min();
+    }
+  else
+    {
+      previous_solution_maximum = this->previous_solutions[0].max();
+      previous_solution_minimum = this->previous_solutions[0].min();
+    }
+
+  // Set minimum value as 1 to prevent overshooting of diffusivity
+  return std::max(previous_solution_maximum - previous_solution_minimum, 1.);
```

Comment:
And this part the (1) should be an argument of the function with a default value of 1. .This way we expose that the function "clips" to a minimum value and we can change that value if need be,.